<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditorWindow</class>
 <widget class="QMainWindow" name="EditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Bot Race - Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="QTabWidget" name="tabWidget_Tiles">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="floor">
         <attribute name="title">
          <string>Floor</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="BotRace::Editor::TileList" name="tileList_Floor"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="wall">
         <attribute name="title">
          <string>Wall</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="BotRace::Editor::TileList" name="tileList_Wall"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="special">
         <attribute name="title">
          <string>Special</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="BotRace::Editor::TileList" name="tileList_Special"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="BotRace::Editor::TileDetailWidget" name="tileDetails" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </widget>
      <widget class="QTabWidget" name="tabWidget_Boards">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>600</width>
         <height>0</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>-1</number>
       </property>
       <property name="documentMode">
        <bool>true</bool>
       </property>
       <property name="tabsClosable">
        <bool>true</bool>
       </property>
       <property name="movable">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Board"/>
    <addaction name="actionLoad_Board"/>
    <addaction name="separator"/>
    <addaction name="actionNew_Scenario"/>
    <addaction name="actionLoad_Scenario"/>
    <addaction name="separator"/>
    <addaction name="actionDraw_Mode"/>
    <addaction name="actionEdit_Mode"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="actionManual"/>
    <addaction name="actionAbout_BoardEditor"/>
    <addaction name="separator"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <widget class="QMenu" name="menuEdit_Board">
    <property name="title">
     <string>Edit Board</string>
    </property>
    <addaction name="actionBoard_properties"/>
    <addaction name="actionUpdate_Laserbeams"/>
    <addaction name="actionAdd_Board_to_Scene"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuEdit_Scenario">
    <property name="title">
     <string>Edit Scenario</string>
    </property>
    <addaction name="actionAdd_Board"/>
    <addaction name="actionRemove_Board"/>
    <addaction name="separator"/>
    <addaction name="actionScenario_Properties"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit_Board"/>
   <addaction name="menuEdit_Scenario"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSave"/>
   <addaction name="actionClose"/>
   <addaction name="separator"/>
   <addaction name="actionNew_Board"/>
   <addaction name="actionLoad_Board"/>
   <addaction name="actionNew_Scenario"/>
   <addaction name="actionLoad_Scenario"/>
   <addaction name="separator"/>
   <addaction name="actionEdit_Mode"/>
   <addaction name="actionDraw_Mode"/>
  </widget>
  <action name="actionNew_Board">
   <property name="text">
    <string>New Board</string>
   </property>
  </action>
  <action name="actionLoad_Board">
   <property name="text">
    <string>Load Board</string>
   </property>
  </action>
  <action name="actionNew_Scenario">
   <property name="text">
    <string>New Scenario</string>
   </property>
  </action>
  <action name="actionLoad_Scenario">
   <property name="text">
    <string>Load Scenario</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/document-save.png</normaloff>:/icons/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/document-save-as.png</normaloff>:/icons/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/document-close.png</normaloff>:/icons/document-close.png</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionManual">
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/help-contents.png</normaloff>:/icons/help-contents.png</iconset>
   </property>
   <property name="text">
    <string>Manual</string>
   </property>
  </action>
  <action name="actionAbout_BoardEditor">
   <property name="text">
    <string>About BoardEditor</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionUpdate_Laserbeams">
   <property name="text">
    <string>Update Laserbeams</string>
   </property>
  </action>
  <action name="actionAdd_Board_to_Scene">
   <property name="text">
    <string>Add Board to Scene</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/configure.png</normaloff>:/icons/configure.png</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionBoard_properties">
   <property name="text">
    <string>Board properties</string>
   </property>
  </action>
  <action name="actionDraw_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/document-edit.png</normaloff>:/icons/document-edit.png</iconset>
   </property>
   <property name="text">
    <string>Draw Mode</string>
   </property>
  </action>
  <action name="actionEdit_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../icons/gameicons.qrc">
     <normaloff>:/icons/edit-select.png</normaloff>:/icons/edit-select.png</iconset>
   </property>
   <property name="text">
    <string>Edit Mode</string>
   </property>
  </action>
  <action name="actionAdd_Board">
   <property name="text">
    <string>Add Board</string>
   </property>
  </action>
  <action name="actionRemove_Board">
   <property name="text">
    <string>Remove Board</string>
   </property>
  </action>
  <action name="actionProperties">
   <property name="text">
    <string>Properties</string>
   </property>
  </action>
  <action name="actionScenario_Properties">
   <property name="text">
    <string>Scenario Properties</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BotRace::Editor::TileList</class>
   <extends>QListWidget</extends>
   <header>tilelist.h</header>
  </customwidget>
  <customwidget>
   <class>BotRace::Editor::TileDetailWidget</class>
   <extends>QWidget</extends>
   <header>tiledetailwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../icons/gameicons.qrc"/>
  <include location="../../icons/gameicons.qrc"/>
  <include location="../../icons/gameicons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionAbout_BoardEditor</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>closeCurrentFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>exit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_Board</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>loadBoard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_Scenario</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>loadScenario()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionManual</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>showManual()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Board</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>newBoard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Scenario</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>newScenario()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSettings</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>showSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_Board_to_Scene</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>addCurrentBoardToScenario()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBoard_properties</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>changeBoardProperties()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDraw_Mode</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>switchToDrawMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdit_Mode</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>switchToEditMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_Board</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>addBoardToScenario()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionScenario_Properties</sender>
   <signal>triggered()</signal>
   <receiver>EditorWindow</receiver>
   <slot>changeScenarioProperties()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>newBoard()</slot>
  <slot>newScenario()</slot>
  <slot>loadBoard()</slot>
  <slot>loadScenario()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>closeCurrentFile()</slot>
  <slot>exit()</slot>
  <slot>showSettings()</slot>
  <slot>about()</slot>
  <slot>showManual()</slot>
  <slot>addBoardToScenario()</slot>
  <slot>changeBoardProperties()</slot>
  <slot>switchToEditMode()</slot>
  <slot>switchToDrawMode()</slot>
  <slot>addCurrentBoardToScenario()</slot>
  <slot>changeScenarioProperties()</slot>
 </slots>
</ui>
